Added test for Chebyshev.

oseikuffuor1 opened the pull request:
Added new regression test for Chebyshev approximation.

oseikuffuor1 committed:
Added test for Chebyshev.

jeanlucf22 commented on tests/CMakeLists.txt:
```diff
@@ -276,6 +276,14 @@ add_test(NAME testLBFGS
 
+add_test(NAME ChebyshevMVP
+         COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/Chebyshev/test.py
+         ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} 9 ${MPIEXEC_PREFLAGS}
```
@oseikuffuor1 can't you run it with fewer MPI tasks? It looks like your PR is failing because of that. I typically have been trying to keep the number of MPI tasks to 4 or less.

Rombur replied:
I think that the limit on the testing machines is 8 processes.

oseikuffuor1 replied:
Thanks, I will try a smaller processor decomposition.

oseikuffuor1 committed:
Changed num procs for Chebyshev test

jeanlucf22 commented:
@oseikuffuor1 
Look at:
https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/mgmol/detail/PR-100/3/pipeline/14
There is a lot of warnings at compile time with clang. Could you fix these? I believe it is just a couple of minor fixes, but that would help spot more important issues when they arise. Thanks!

oseikuffuor1 committed:
Cleanup some CI warnings.

oseikuffuor1 commented:
@jeanlucf22 the latest commit should fix the warnings I introduced. I also cleaned up one in GramMatrix.h by changing int dim_ to unsigned int. It built cleanly for me, but I anticipate there may be other warnings depending on how it is used elsewhere, and how strict the compiler is. I will wait to see what the checks say, and I will address any issues accordingly.
